### ISSUE TITLE
`canonicalizePath` throws error on non-existing path

### ISSUE THREAD

**Issue by parsonsmatt** [haskell/directory#45]
I came across this in idris-lang/Idris-dev#2982. Steps to reproduce (running Ubuntu 14.04)

```
λ ~/Projects/ stack ghci --package directory
Run from outside a project, using implicit global project config
Using resolver: lts-5.1 from implicit global project's config file: /home/matt/.stack/global-project/stack.yaml
Configuring GHCi with the following packages: 
GHCi, version 7.10.3: http://www.haskell.org/ghc/  :? for help
Ok, modules loaded: none.
Prelude> import System.Directory
Prelude System.Directory> canonicalizePath "."
"/home/matt/Projects"
Prelude System.Directory> canonicalizePath "./asdf"
*** Exception: ./asdf: canonicalizePath: does not exist (No such file or directory)
```

The documentation says:

> The input path does not have to point to an existing file or directory.

and it does seem like there's some error checking on that.

**Comment by Rufflewind** [comment]
The [bug](https://github.com/haskell/directory/issues/23) was fixed in directory-1.2.3.0.  Can you try a newer version and see if the bug still occurs?

**Comment by parsonsmatt** [comment]
Thanks! Sorry for not noticing that, I should have checked the versions more carefully. The behavior is working for me with that version.
